core/conversion/evaluators/aten.cpp

```diff
@@ -270,7 +270,11 @@ auto aten_registrations TORCHTRT_UNUSED =
                  if (tensor_var.isITensor()) {
                    auto tensor = tensor_var.ITensor();
                    if (ctx->input_is_dynamic) {
-                     return dynamic_size_layer(ctx, n, args);
+                     if (ctx->settings.allow_shape_tensors) {
+                       return dynamic_size_layer(ctx, n, args);
+                     } else {
+                       LOG_WARNING("There may be undefined behavior using dynamic shape and aten::size ");
```

Comment:
Is this undefined behavior due to using aten::size with dynamic shape without shape tensors or the other way around. Might need rewording